src/balances/balances.controller.spec.ts

```diff
@@ -1,22 +1,68 @@
 import { Test, TestingModule } from '@nestjs/testing';
-import { SafeTransactionModule } from '../services/safe-transaction/safe-transaction.module';
-import { BalancesController } from './balances.controller';
-import { BalancesService } from './balances.service';
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import axios from 'axios';
+import { BalancesModule } from './balances.module';
 
-describe('BalancesController', () => {
-  let controller: BalancesController;
+jest.mock('axios');
+const axiosMock = axios as jest.Mocked<typeof axios>;
+
+describe('AppController (e2e)', () => {
+  let app: INestApplication;
 
   beforeEach(async () => {
-    const module: TestingModule = await Test.createTestingModule({
-      imports: [SafeTransactionModule],
-      controllers: [BalancesController],
-      providers: [BalancesService],
+    jest.clearAllMocks();
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [BalancesModule],
     }).compile();
+    app = moduleFixture.createNestApplication();
+    await app.init();
+  });
+
+  it(`GET /balances`, async () => {
+    const chainId = 1;
+    const safeAddress = '0x0000000';
+    const safeTransactionServiceBalancesResponse = {
+      data: [{ tokenAddress: '0x000000', balance: '1' }],
+    };
+    const safeConfigServiceResponse = {
+      data: {
+        chainId: '1',
+        chainName: 'mainnet',
+        transactionService: 'https://test.safe.transaction.service',
+        vpcTransactionService: 'https://test.safe.transaction.service',
+      },
+    };
```

Comment:
We can probably extract this into a separate folder that contains mocked responses/json.

We can do this as soon as we start testing multiple scenarios 👍

Reply:
Yes, cool. I have mixed opinions about this to be honest, could be good to have a folder for fixtures but I think we need to be cautious with that as it could increase coupling between test suites (from [my fav testing best practices repository :)](https://github.com/goldbergyoni/javascript-testing-best-practices#-%EF%B8%8F16-dont-foo-use-realistic-input-data))

Reply:
I also think it could be useful to generate test data using a tool which generates different inputs in each test execution, like [Faker](https://www.npmjs.com/package/@faker-js/faker) (it even has an [Eth address generator](https://fakerjs.dev/api/finance.html#ethereumAddress) 😄)

Reply:
Regarding Faker we do use it on our Django services and I was thinking about introducing it but in another PR!